kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -337,20 +344,19 @@ define([
                 }
                 changeMsg.newDisplayValue = model.displayValue;
                 changeMsg.newValue = newValue;
+                channel.emit('changed', changeMsg);
             } else if (result.diagnosis === Constants.DIAGNOSIS.REQUIRED_MISSING) {
                 model.value = spec.data.nullValue;
+                channel.emit('changed', changeMsg);
             }
-            channel.emit('changed', changeMsg);
-            channel.emit('validation', {
-                errorMessage: result.errorMessage,
-                diagnosis: result.diagnosis,
-            });
+            channel.emit('validation', result);
```

Comment:
I don't much like the duplicate `channel.emit('changed')` bit, but it shouldn't emit `changed` when there's an error. IMO it's clean enough, but I bet I'm missing something obvious.